Octokit.Tests.Integration/Clients/RepositoryInvitationsClientTests.cs

```diff
@@ -189,7 +188,7 @@ public class TheGetAllForCurrentMethod
             RepositoryContext[] contexts = null;
             try
             {
-                contexts = await Task.WhenAll(repoNames.Select(x => github.CreateRepositoryContext(new NewRepository(x))));
+                contexts = repoNames.Select(x => github.CreateRepositoryContext(new NewRepository(x)).Result).ToArray();
```

Comment:
Do you have more details on why you removed the parallel async code in these tests?

Reply:
@gdziadkiewicz just checking you saw my message asking why the `WhenAll` was removed from here?

Reply:
@ryangribble  Sorry for the delay. With the `WhenAll` version I am getting `NullReferenceException`  with stacktrace:
```
System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeEnumHelper(String value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.SimpleJson.DeserializeObject(String json, Type type, IJsonSerializerStrategy jsonSerializerStrategy)
   at Octokit.SimpleJson.DeserializeObject[T](String json, IJsonSerializerStrategy jsonSerializerStrategy)
   at Octokit.Internal.SimpleJsonSerializer.Deserialize[T](String json)
   at Octokit.Internal.JsonHttpPipeline.DeserializeResponse[T](IResponse response)
   at Octokit.Connection.<Run>d__56`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Octokit.ApiConnection.<Post>d__22`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Octokit.RepositoriesClient.<Create>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Octokit.Tests.Integration.Helpers.GithubClientExtensions.<CreateRepositoryContext>d__2.MoveNext() in C:\Repos\octokit.net\Octokit.Tests.Integration\Helpers\GithubClientExtensions.cs:line 24
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at RepositoryInvitationsClientTests.TheGetAllForCurrentMethod.<ReturnsCorrectCountOfInvitationsWithStart>d__1.MoveNext() in C:\Repos\octokit.net\Octokit.Tests.Integration\Clients\RepositoryInvitationsClientTests.cs:line 191
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

Reply:
@ryangribble Sorry for the delay. The `WhenAll` version throws:
```
System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeEnumHelper(String value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.PocoJsonSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(Object value, Type type)
   at Octokit.SimpleJson.DeserializeObject(String json, Type type, IJsonSerializerStrategy jsonSerializerStrategy)
   at Octokit.SimpleJson.DeserializeObject[T](String json, IJsonSerializerStrategy jsonSerializerStrategy)
   at Octokit.Internal.SimpleJsonSerializer.Deserialize[T](String json)
   at Octokit.Internal.JsonHttpPipeline.DeserializeResponse[T](IResponse response)
   at Octokit.Connection.<Run>d__56`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Octokit.ApiConnection.<Post>d__22`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Octokit.RepositoriesClient.<Create>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Octokit.Tests.Integration.Helpers.GithubClientExtensions.<CreateRepositoryContext>d__2.MoveNext() in C:\Repos\octokit.net\Octokit.Tests.Integration\Helpers\GithubClientExtensions.cs:line 24
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at RepositoryInvitationsClientTests.TheGetAllForCurrentMethod.<ReturnsCorrectCountOfInvitationsWithStart>d__1.MoveNext() in C:\Repos\octokit.net\Octokit.Tests.Integration\Clients\RepositoryInvitationsClientTests.cs:line 191
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

Reply:
OK I think this might be #1683 causing issues when running in parallel.  It's funny because it works OK for me - it's all down to timing I guess!

Im not a fan of using `.Result` though since it's a real hack around `async` code...  Maybe this can be written as a `foreach (var repoName in repoNames)` loop that creates the contexts one by one and adds them to a list.

Reply:
Changed it.